src/typespec.h

```diff
@@ -81,6 +81,13 @@ struct TypeSpec
 
     /// Get number of bytes required to store the field for a single point
     int size() const { return elsize*count; }
+
+    bool operator==(const TypeSpec& other) const
```

Comment:
One line of docs would be worth it.  Perhaps "Check that two TypeSpecs are exactly equal"